lib/ansible/modules/cloud/amazon/elb_application_lb_facts.py

```diff
@@ -0,0 +1,260 @@
+                                      ec2_argument_spec, get_aws_connection_info)
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError, NoCredentialsError
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def get_load_balancer_attributes(connection, module, load_balancer_arn):
+
+    try:
+        load_balancer_attributes = boto3_tag_list_to_ansible_dict(connection.describe_load_balancer_attributes(LoadBalancerArn=load_balancer_arn)['Attributes'])
+    except ClientError as e:
+        module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+
+    # Replace '.' with '_' in attribute key names to make it more Ansibley
+    for k, v in load_balancer_attributes.items():
+        load_balancer_attributes[k.replace('.', '_')] = v
+        del load_balancer_attributes[k]
+
+    return load_balancer_attributes
+
+
+def get_load_balancer_tags(connection, module, load_balancer_arn):
+
+    try:
+        return boto3_tag_list_to_ansible_dict(connection.describe_tags(ResourceArns=[load_balancer_arn])['TagDescriptions'][0]['Tags'])
+    except ClientError as e:
+        module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+
+
+def list_load_balancers(connection, module):
+
+    load_balancer_arns = module.params.get("load_balancer_arns")
+    names = module.params.get("names")
+
+    try:
+        load_balancer_paginator = connection.get_paginator('describe_load_balancers')
+        if not load_balancer_arns and not names:
+            load_balancers = load_balancer_paginator.paginate().build_full_result()
+        if load_balancer_arns:
+            load_balancers = load_balancer_paginator.paginate(LoadBalancerArns=load_balancer_arns).build_full_result()
+        if names:
+            load_balancers = load_balancer_paginator.paginate(Names=names).build_full_result()
+    except ClientError as e:
+        if e.response['Error']['Code'] == 'LoadBalancerNotFound':
+            module.exit_json(load_balancers=[])
+        else:
+            module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+    except NoCredentialsError as e:
+        module.fail_json(msg="AWS authentication problem. " + e.message, exception=traceback.format_exc())
+
+    # Get the attributes and tags for each target group
+    for load_balancer in load_balancers['LoadBalancers']:
+        load_balancer.update(get_load_balancer_attributes(connection, module, load_balancer['LoadBalancerArn']))
+        load_balancer['tags'] = get_load_balancer_tags(connection, module, load_balancer['LoadBalancerArn'])
+
+    # Turn the boto3 result in to ansible_friendly_snaked_names
+    snaked_load_balancers = [camel_dict_to_snake_dict(load_balancer) for load_balancer in load_balancers['LoadBalancers']]
+
+    module.exit_json(load_balancers=snaked_load_balancers)
```

Comment:
For the output, would it be possible to also return these in a dictionary by ARN and name? So you'd have output along the lines of:

```
{"load_balancers": [a list like you have now],
 "load_balancers_by_arn": {"arn:aws:alb:....": {ALB object}},
 "load_balancers_by_name": {"fooBarBalancer": {ALB object}}
}
```

That would make it easier to get a particular ALB instead of users having to do list indexing and such.